### DISCUSSION TITLE
Can't find a DNS resolver config.

### DISCUSSION THREAD

**Discussion by PakToPack** [roadrunner-server/roadrunner#1646] [Q&A] [answered]
So, what's the problem:
When using vanilla php I get my nginx+apache set up to use `dnsmasq`

`dnsmasq` cashes the dns ip address and removes query time to resolve ip.

It works with my system, it works with my clear-code php and apache BUT **somehow it doesn't work with RoadRunner**

**_All configs of dnsmasq and php are the same_**. Where can I set roadrunner DNS resolver to DNSMasq and not any other NS Service?

**Comment by rustatian** [comment]
Hey @PakToPack 👋🏻 
RR is not a web-server or `DNS` forwarder. You may use address like `http://my-address` and RR automatically resolve it and bind to it.

**Reply by PakToPack** [reply]
> Hey @PakToPack 👋🏻 
> 
> RR is not a web-server or `DNS` forwarder. You may use address like `http://my-address` and RR automatically resolve it and bind to it.

Yep. But even tho, when I use apache to run my PHP worker: all fine, DNS resolves very fast. When I use roadrunner: I get timeouts and leaks just because of overlapping of DNS resolve queries from CURL. Idk how to fix that to continue using RR.

That's just a fact. 1:1 same code runs with RoadRunner: DNSMasq idling. Runs with apache+nginx: DNSMasq is loaded heavily and all fine, no timeouts.

So with roadrunner I can handle like 100 rps, apache+nginx - 3.000 rps.

/etc/resolve.conf contains just this:
nameserver 127.0.0.1

Can I set this value in RR or sum? Bcz it's strange that when I use RR as web server it ignores system name server.

**Reply by rustatian** [reply]
So, you're using `DNSMasq` in PHP code with cURL, right?

**Reply by rustatian** [reply]
Is there are any logs from the:
- `dnsmasqd` daemon
- RR debug logs
- `dnsmasqd` configuration
- Have you checked the correctness of the configuration for the max number of the concurrent connection?

**Reply by PakToPack** [reply]
> Is there are any logs from the:
> 
> - `dnsmasqd` daemon
> 
> - RR debug logs
> 
> - `dnsmasqd` configuration
> 
> - Have you checked the correctness of the configuration for the max number of the concurrent connection? 

DNS Masq not getting any requests, so no logs there

RR is not debugging, it's just running like usual - no logs there

dnsmasq is in default config, no errors there

Max number of concurrent connections for apache is like 1500-2000/s? So why RR is like 100 rps with same exact code? That's the original question. I don't understand why this happens.

It's like:
The code works, but it feels like RR has it's own dns settings that I don't see

**Reply by rustatian** [reply, marked as answer]
RR doesn't have its own dns settings at all. Since dnsmasq is not getting any requests - check your application code.
Note that RR does not kill the workers when the request is finished. You should not store state between requests.
You can use `max_jobs: 1` option to emulate `nginx`/`apach` behavior (destroy worker after the request is done).
